archspec/cpu/detect.py

```diff
@@ -206,11 +206,26 @@ def host():
     # Get a list of possible candidates for this micro-architecture
     candidates = compatible_microarchitectures(info)
 
+    # Sorting criteria for candidates
+    def sorting_fn(item):
+        return len(item.ancestors), len(item.features)
+
+    # Get the best generic micro-architecture
+    generic_candidates = [c for c in candidates if c.vendor == "generic"]
+    best_generic = sorted(generic_candidates, key=sorting_fn, reverse=True)[0]
+
+    # Filter the candidates to be descendant of the best generic candidate.
+    # This is to avoid that the lack of a niche feature that can be disabled
+    # from e.g. BIOS prevents detection of a reasonably performant architecture
+    candidates = [c for c in candidates if c > best_generic]
```

Comment:
I'm not sure why this is needed, but perhaps I'm too focused on the `haswell` without `aes` case from #40...

In that particular case, `haswell` would not be part of `candidates` at all, would it?
If that's correct, do you have an example in mind where this filtering does make sense?

Reply:
This is needed especially there. We want to remove from the candidates all the Intel architectures that are less than `x86_64_v3` (i.e. we don't want to consider uarchs that don't support AVX2, since `x86_64_v3` has been detected)

Reply:
Right, understood.